### PR TITLE
Set up ESLint for client and server

### PR DESCRIPTION
**Pull request by Archi-shaw**

### CODE REVIEW

**Comment by Archi-shaw** [comment]
@mendacium-a11y  Please review it

Fixes issue #14

**Comment by Archi-shaw** [comment]
@mendacium-a11y Please review it

Fixes issue #14

**Comment by mendacium-a11y** [comment]
pls refrain from tagging me repeatedly in the comments and ensure from next time you have a PR description as well

File: README.md [hunk 1]
```diff
@@ -51,6 +51,39 @@ This section should list any major frameworks/libraries used to bootstrap your p
...
+#### Frontend (`client/`):
+
+- **Configuration File**: `.eslintrc.cjs`
+- **Ignored Files**: `.eslintignore`
+  - `node_modules/`: Contains third-party dependencies.
+  - `build/`: Contains the bundled output generated by the build tool.
+  - `public/`: Includes static files like images and fonts.
+  - `.env`: Stores sensitive environment variables.
+  - `*.test.js`: Optional—can be included or excluded based on your setup.
+
```
Comment (mendacium-a11y) [inline README.md]: i dont think all this needed in the read me, pls remove this

File: README.md [hunk 2]
```diff
@@ -51,6 +51,39 @@ This section should list any major frameworks/libraries used to bootstrap your p
...
+  cd server
+
+  npm run lint
+
+ **Frontend**:
+  ```bash
+  cd client
+  
+  npm run lint
+
```
Comment (mendacium-a11y) [inline README.md:54]: pls remove all of this as well as in the future a precommit hook will be created that would run the linter first

File: client/.eslintrc.cjs [hunk 3]
```diff
@@ -1,18 +1,43 @@
...
       'warn',
       { allowConstantExport: true },
     ],
+    "react-refresh/only-export-components": "off",
+    "jsx-a11y/anchor-is-valid": "off",
+    "no-useless-escape": "off",
+    '@typescript-eslint/no-explicit-any': 'off' ,
+    'react/prop-types': ['off'],  
+    'indent': ['error', 2],      
+    'react/jsx-filename-extension': ['warn', { extensions: ['.jsx', '.tsx'] }],
```
Comment (mendacium-a11y) [inline client/.eslintrc.cjs]: change this to off

File: server/.eslintrc.json
Comment (mendacium-a11y) [inline server/.eslintrc.json:1]: in the issue it is specified to use an eslintrc.cjs file, pls update

File: server/.eslintrc.json [hunk 4]
```diff
@@ -1,15 +1,24 @@
...
-        "es2021": true
+      "node": true,
+      "es2021": true
     },
-    "extends": "standard",
+    "extends": [
+      "standard",
+      "plugin:@typescript-eslint/recommended"
+    ],
+    "parser": "@typescript-eslint/parser",
```
Comment (mendacium-a11y) [inline server/.eslintrc.json]: no need for this rn as the backend is in vanilla js not ts

File: client/.eslintrc.cjs [hunk 5]
```diff
@@ -1,18 +1,43 @@
...
     'react-refresh/only-export-components': [
       'warn',
       { allowConstantExport: true },
     ],
+    "react-refresh/only-export-components": "off",
+    "jsx-a11y/anchor-is-valid": "off",
+    "no-useless-escape": "off",
+    '@typescript-eslint/no-explicit-any': 'off' ,
+    'react/prop-types': ['off'],  
+    'indent': ['error', 2],
```
Comment (mendacium-a11y) [inline client/.eslintrc.cjs]: update this to 4

File: server/.eslintrc.cjs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  ],
+  parserOptions: {
+      ecmaVersion: "latest",
+      sourceType: "module",
+  },
+  rules: {
+      "no-useless-escape": "off",
+      "indent": ["error", 4],
+      "no-console": "off",
+      "@typescript-eslint/no-unused-vars": "off",
```
Comment (mendacium-a11y) [inline server/.eslintrc.cjs]: pls remove ts based dependencies, backend is js based

File: server/package.json [hunk 7]
```diff
@@ -19,10 +19,14 @@
   },
   "type": "module",
   "devDependencies": {
-    "eslint": "^8.57.0",
+    "@typescript-eslint/eslint-plugin": "^8.18.0",
+    "@typescript-eslint/parser": "^8.18.0",
```
Comment (mendacium-a11y) [inline server/package.json]: pls remove ts based dependencies, backend is js based

**Pull request merged**